src/workflows/airqo_etl_utils/airqo_api.py

```diff
@@ -155,15 +159,17 @@ def get_thingspeak_read_keys(self, devices: list) -> dict:
                 )
 
         response = self.__request("devices/decrypt/bulk", body=body, method="post")
-        # TODO Check for when no data is returned
-        decrypted_keys = response.get("decrypted_keys", [])
 
-        return {
-            int(entry["device_number"]): entry["decrypted_key"]
-            for entry in decrypted_keys
-        }
+        if response:
+            decrypted_keys = response.get("decrypted_keys", [])
+            return {
+                int(entry["device_number"]): entry["decrypted_key"]
+                for entry in decrypted_keys
+            }
+        # TODO Find a better way to do better handling vs returning an empty object.
```

Comment:
Can this suggested better way be implemented in this PR? Or does it require more time to analyse? @NicholasTurner23

Reply:
I wanted to make sure I find all the method calls/instances before I can change it all. But will still be a work in progress for the entire workflows code base as we switch to static typing. Making sure that methods either return the required payload or an error message in case something goes wrong.

Reply:
Okay, thanks @NicholasTurner23 ! Makes sense!